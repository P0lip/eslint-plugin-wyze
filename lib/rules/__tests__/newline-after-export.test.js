import { CLIEngine, RuleTester } from 'eslint'
import rule from '../newline-after-export'

const parserOptions = {
  ecmaVersion: 2017,
  sourceType: 'module',
}
const message = 'Expected newline after export statement.'

describe('newline-after-export', () => {
  it('passes RuleTester', () => {
    new RuleTester().run('newline-after-export', rule, {
      valid: [
        {
          code:
          `
          export const a = 1;
          export const b = 2;

          export default { a, b };
          `,
          parserOptions,
        },
      ],
      invalid: [
        {
          code:
          `
          export const a = 1;
          const b = 1;
          `,
          errors: [
            { message },
          ],
          parserOptions,
        },
      ],
    })
  })

  it('applies fixes', () => {
    const cli = new CLIEngine({
      cwd: __dirname,
      fix: true,
      parserOptions,
      plugins: [
        'wyze',
      ],
      rulePaths: [ '..' ],
      rules: {
        'wyze/newline-after-export': 'error',
      },
      useEslintrc: false,
    })
    const fixture = name => [ `fixtures/newline-after-export/${name}.js` ]
    const title = name => `newline-after-export applies fixes to ${name} export`

    expect(cli.executeOnFiles(fixture('default')))
      .toMatchSnapshot(title('default'))

    expect(cli.executeOnFiles(fixture('named')))
      .toMatchSnapshot(title('named'))
  })
})
