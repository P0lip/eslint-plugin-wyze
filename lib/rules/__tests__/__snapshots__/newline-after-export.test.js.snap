// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newline-after-export applies fixes to default export 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "/Users/neil/Projects/personal/eslint-plugin-wyze/lib/rules/__tests__/fixtures/newline-after-export/default.js",
      "messages": Array [],
      "output": "
export default 1

const b = 2
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;

exports[`newline-after-export applies fixes to named export 1`] = `
Object {
  "errorCount": 0,
  "results": Array [
    Object {
      "errorCount": 0,
      "filePath": "/Users/neil/Projects/personal/eslint-plugin-wyze/lib/rules/__tests__/fixtures/newline-after-export/named.js",
      "messages": Array [],
      "output": "
export const a = 1

const b = 2
",
      "warningCount": 0,
    },
  ],
  "warningCount": 0,
}
`;
